include/lsst/afw/image/Exposure.h

```diff
@@ -444,6 +444,24 @@ class Exposure {
      */
     Exposure getCutout(lsst::geom::Point2D const& center, lsst::geom::Extent2I const& size) const;
 
+    /**
+     * Return an Exposure that is a small cutout of the original.
+     *
+     * This is distinguished from subset() by allowing the box to extend off the image, and padding with
+     * NaN and mask NO_DATA in that case.
+     *
+     * @param box Pixel box to cut from this exposure.
+     *
+     * @return An Exposure of the requested box to within half a pixel in either dimension. Pixels past the
+     *     edge of the original exposure will equal @ref math::edgePixel(image::detail::MaskedImage_tag)
+     *     "math::edgePixel<MaskedImageT>", with their Mask set to NO_DATA.
+     *
+     * @throws lsst::pex::exceptions::InvalidParameterError Thrown if the center of the box
+     *     falls outside this Exposure.
+     */
+    Exposure getCutout(lsst::geom::Box2D const& box) const;
+    Exposure getCutout(lsst::geom::Box2I const& box) const;
```

Comment:
For me the question is whether there is something extra `getCutout` can do with a `Box2D` that it couldn't do with a `Box2I` (e.g. set some metadata field on return).  If it's just a convenience for some way of getting a `Box2I` from a `Box2D` and then calling `getCutout`, then I don't think we want the overload.

Reply:
Thanks, yes, I meant to wrap it in `@{ @}`.

I added the `Box2D` version because that seemed like the first thing a user would want (cutout of a generic box), while the `Box2I` version is what you'd do with e.g. `footprint.getBBox()`. I figured it would be helpful for users to not have to remember to do `geom.ceil(box2d)`, as I think it's challenging for users to remember how to convert between our various primitives. I could add the box2d version as just a python (or pybind11) interface, if that would help?

Reply:
> I could add the box2d version as just a python (or pybind11) interface, if that would help?

I don't think it would. The issue is that the correct conversion from `Box2D` to `Box2I` is context-dependent no matter which language you call it in. I assume `Box2I(Box2D, EdgeHandlingEnum)` is accessible from Python (possibly with a string replacing the enum)?

Reply:
I've removed the Box2D overload. We'll see if any users want that functionality.